components/reference-data/src/test/java/org/datacleaner/beans/transform/RemoveDictionaryMatchesTransformerTest.java

```diff
@@ -74,31 +74,30 @@ public void testJobTitleScenarioRemovedMatchesAsString() throws Throwable {
         assertEquals("Software Engineer", transformer.transform("Senior Software Engineer")[0]);
         assertEquals("Senior", transformer.transform("Senior Software Engineer")[1]);
         
-        assertEquals("Designer", transformer.transform("  Lead   Designer  ")[0]);
+        assertEquals("    Designer  ", transformer.transform("  Lead   Designer  ")[0]);
         assertEquals("Lead", transformer.transform("  Lead   Designer  ")[1]);
         
-        assertEquals("Software Engineer", transformer.transform("Software  Engineer")[0]);
+        assertEquals("Software  Engineer", transformer.transform("Software  Engineer")[0]);
         assertEquals("", transformer.transform("Software  Engineer")[1]);
         
         assertEquals("Guru of employees", transformer.transform("Principal Senior Lead Guru of Junior employees")[0]);
-        assertEquals("Principal Senior Lead Junior", transformer.transform("Principal Senior Lead Guru of Junior employees")[1]);
+        assertEquals("Principal Junior Senior Lead", transformer.transform("Principal Senior Lead Guru of Junior employees")[1]);
    
     }
     public void testJobTitleScenarioRemovedMatchesAsList() throws Throwable {
-    
         transformer._removedMatchesType= RemovedMatchesType.LIST;
         
         assertEquals("Software Engineer", transformer.transform("Senior Software Engineer")[0]);
         assertEquals("[Senior]", transformer.transform("Senior Software Engineer")[1].toString());
         
-        assertEquals("Designer", transformer.transform("  Lead   Designer  ")[0]);
+        assertEquals("    Designer  ", transformer.transform("  Lead   Designer  ")[0]);
         assertEquals("[Lead]", transformer.transform("  Lead   Designer  ")[1].toString());
         
-        assertEquals("Software Engineer", transformer.transform("Software  Engineer")[0]);
+        assertEquals("Software  Engineer", transformer.transform("Software  Engineer")[0]);
         assertEquals("[]", transformer.transform("Software  Engineer")[1].toString());
         
         assertEquals("Guru of employees", transformer.transform("Principal Senior Lead Guru of Junior employees")[0]);
-        assertEquals("[Principal, Senior, Lead, Junior]", transformer.transform("Principal Senior Lead Guru of Junior employees")[1].toString());
+        assertEquals("[Principal, Junior, Senior, Lead]", transformer.transform("Principal Senior Lead Guru of Junior employees")[1].toString());
```

Comment:
Damn. I don't think this side-effect is really good. It used to be a good thing that you could rely on having the same order of dictionary terms in the output as in the input.

Maybe this calls for an alteration of the strategy - just thinking aloud here:
 * Maybe we should only do the "replace inline" stuff on multi-word dictionary terms.
 * And then we could retain the old logic of tokenizing the input first and looking up every individual token in the order of the input string.